Shopipy.ProductManagement/Controllers/ProductController.cs

```diff
@@ -29,15 +35,30 @@ public async Task<IActionResult> CreateProduct(ProductRequestDTO dto, int busine
     }
 
     [HttpGet]
-    public async Task<IActionResult> GetAllProducts(int businessId, [FromQuery] int? top = null, [FromQuery] int? skip = null)
+    public async Task<IActionResult> GetAllProducts(int businessId, [FromQuery] int? top = null, [FromQuery] int? skip = null, [FromQuery] int? categoryId = null)
     {
-        var products = await _productService.GetAllProductsOfBusinessAsync(businessId, top, skip);
-        var count = await _productService.GetProductCountAsync(businessId);
+        if (categoryId.HasValue)
+        {
+            var category = await categoryService.GetCategoryByIdAsync(categoryId.Value);
+            if (category == null) return NotFound("Category does not exist");
+
+            var products = await _productService.GetAllProductsOfCategoryBusinessAsync(businessId, categoryId.Value, top, skip);
+            var count = await _productService.GetProductCountCategoryAsync(businessId, categoryId.Value);
+
+            return Ok(new PaginationResultDto<ProductResponseDTO>
+            {
+                Data = products.Select(_mapper.Map<ProductResponseDTO>),
+                Count = count
+            });
+        }
+
+        var allProducts = await _productService.GetAllProductsOfBusinessAsync(businessId, top, skip);
+        var allCount = await _productService.GetProductCountAsync(businessId);
 
         return Ok(new PaginationResultDto<ProductResponseDTO>
         {
-            Data = products.Select(_mapper.Map<ProductResponseDTO>),
-            Count = count
+            Data = allProducts.Select(_mapper.Map<ProductResponseDTO>),
+            Count = allCount
```

Comment:
Since you are already making changes to this part, could you get rid of this `if` and fetch products by passing optional `categoryId` to the service?

Reply:
sure. was just copying the logic from services

Reply:
fixed